### PR TITLE
Task-24: Google Auth: Front-End / Back-End

### PR DESCRIPTION
**Pull request by melonholy**

### CODE REVIEW

File: config/webpack.config.dev.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const path = require('path');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
+const Dotenv = require('dotenv-webpack');
```
Comment (universezzz) [inline config/webpack.config.dev.js:3]: `Dotenv => dotEnv`

File: src/containers/auth/login/index.js [hunk 2]
```diff
@@ -49,6 +50,10 @@ class Login extends Component {
     }
   };
 
+  responseGoogle = (response) => {
```
Comment (universezzz) [inline src/containers/auth/login/index.js]: `responseGoogle => handleGoogleResponse`

File: src/containers/auth/login/index.js [hunk 3]
```diff
@@ -98,6 +111,7 @@ const mapStateToProps = state => ({});
 
 const mapDispatchToProps = {
   loginRequest,
+  gooogleLoginRequest
```
Comment (universezzz) [inline src/containers/auth/login/index.js]: WTF? gooogleLoginRequest => googleLoginRequest

File: src/sagas/auth.js [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
 import axios from 'axios';
 import types from '../actions/types';
 import * as actions from '../actions/auth';
 import setAuthToken from '../services/setAuthToken';
 
+const googleLoginRequest = res => {
+  return axios
+    .post('http://localhost:8080/api/users/google/auth', res)
+    .then(res => {
+      const decoded = jwt_decode( res.data.token);
```
Comment (universezzz) [inline src/sagas/auth.js]: pls, use shift+alt+f

File: src/sagas/auth.js [hunk 5]
```diff
@@ -1,10 +1,22 @@
...
 import jwt_decode from 'jwt-decode';
 import axios from 'axios';
 import types from '../actions/types';
 import * as actions from '../actions/auth';
 import setAuthToken from '../services/setAuthToken';
 
+const googleLoginRequest = res => {
+  return axios
+    .post('http://localhost:8080/api/users/google/auth', res)
+    .then(res => {
```
Comment (universezzz) [inline src/sagas/auth.js]: you can destruct response
`const {token} = res.data;

File: src/sagas/auth.js [hunk 6]
```diff
@@ -1,10 +1,22 @@
...
+import { call, put, takeLatest, takeEvery, all } from 'redux-saga/effects';
 import jwt_decode from 'jwt-decode';
 import axios from 'axios';
 import types from '../actions/types';
 import * as actions from '../actions/auth';
 import setAuthToken from '../services/setAuthToken';
 
+const googleLoginRequest = res => {
+  return axios
+    .post('http://localhost:8080/api/users/google/auth', res)
```
Comment (universezzz) [inline src/sagas/auth.js]: rewrite route to 'api/users/google/auth'

**Review by universezzz — COMMENTED** [review]
Review completed

File: src/sagas/auth.js [hunk 7]
```diff
@@ -1,10 +1,20 @@
...
+const googleLoginRequest = res => {
+  return axios.post('api/users/google/auth', res).then(res => {
+    const { token } = res.data;
+    const decoded = jwt_decode(token);
+    localStorage.setItem('token', token);
+    setAuthToken(token);
+    return decoded;
+  });
+};
+
```
Comment (slavasgod) [inline src/sagas/auth.js:6]: Migrate this to services

File: .env [hunk 8]
```diff
@@ -0,0 +1 @@
+CLIENT_ID=2324024081-2nk8c6amog25npupplbmaneaj35ot6c7.apps.googleusercontent.com
```
Comment (OlegPanasyuk) [inline .env:1]: It's a mistake to store .env file in public repository. Convert it to .env.example file. It should be better to give us a token by Skype chat.

**Pull request merged**